base/channels.jl

```diff
@@ -421,6 +421,20 @@ isready(c::Channel) = n_avail(c) > 0
 n_avail(c::Channel) = isbuffered(c) ? length(c.data) : length(c.cond_put.waitq)
 isempty(c::Channel) = isbuffered(c) ? isempty(c.data) : isempty(c.cond_put.waitq)
 
+"""
+    willblockput(c::Channel)
+
+Returns `true` if a [`put!`](@ref) on a [`Channel`](@ref) will block
+"""
+willblockput(c::Channel) =  isbuffered(c) ? (n_avail(c) == c.sz_max) : (length(c.cond_take.waitq) == 0)
```

Comment:
```suggestion
willblockput(c::Channel) =  isbuffered(c) ? (n_avail(c) == c.sz_max) : (length(c.cond_take.waitq) != 0)
```

but I might be mistaken. This should be tested.

Reply:
1. Changing to putwillblock is a good suggestion. 

2. isfull() is used for a circular buffer in DataStructues.jl and I think that’s where the feeling that a different name should come from. I can’t say I have a strong opinion on this one way or the other.